docs/Specifications/Workgroup Creation & Management/Interfaces and schemas for the main logical input objects and resolvers.md

```diff
@@ -0,0 +1,233 @@
+
+The current version of this document is hugely influenced by the Radish34 demo use-case and we expect it to be gradually becoming more generic with the addition of new use-cases and abstracting Radish34 into Baseline.
+
+## Logical Inputs
+
+The logical inputs in a workgroup creation and management are input parameters that have to be provided to be provided from the workgroup management body in order for certain actions to be done. The concrete inputs required by the various actions are yet to be defined in another specification. This document will highlight all the main logical inputs - Keystore, organisation config, zero-knowledge circuit verifier key, and contract artefacts.
+
+## Resolvers
+
+The resolvers are objects/classes designed to allow for these logical objects to be inputted in different formats and through different mediums and be transformed(resolved) into a format(schema) that the baseline tools will be able to work with.
+
+The resolvers are a point of extensibility allowing for additional community-created resolvers to be developed as long as they adhere to the specific logical input resolver interface.
+
+# Keystore
+
+## Keystore as logical input
+
+The keystores are containers of signing keys. These keys provide certain abilities for the workgroup administrator. Most commonly these are needed for signing transactions - deploying contracts, performing administrative transactions & more. 
+
+## Keystore resolver interface
+
+The keystore resolver defines an abstract interface to be implemented. This interface governs how keystores are to be inputted in the baseline tooling and what the expected outputs are.
+
+Any keystore resolver should be a class that adheres to the following interface:
+
+    interface KeystoreResolver {
+    	resolve(extraParams) : Promise<ethers.Signer>
+    }
+
+Invocation of the `resolve` method of the resolver should return a promise returning an object adhering to the `ethers.Signer` ([https://github.com/ethers-io/ethers.js/blob/master/src.ts/abstract-signer.ts#L12](https://github.com/ethers-io/ethers.js/blob/master/src.ts/abstract-signer.ts#L12)). In the future, we might decide to change to even more generic signer or web3 instance.
+
+The `extraParams` are additional resolver-specific customization params passed as an input that might be needed for the resolver to work. The idea behind `extraParams` is to allow for further customizations of the resolver behavior. An example of extra params might be headers objects for a REST API based resolver.
+
+## Inbuilt Baseline Keystore resolvers
+
+### KeystorePathResolver
+
+As part of the Radish34, a `KeystorePathResolver` is to be developed. This resolver would adhere to the `KeystoreResolver` interface.
```

Comment:
Is this part of Radish34 though? Or is it part of Baseline?

Feels like any of this effort would fall under generic baseline packaging instead of radish34.

Reply:
Yep you are right.